eks_fargate/datadog_checks/eks_fargate/eks_fargate.py

```diff
@@ -1,31 +1,141 @@
+except ImportError:
+
+    def get_config(key):
+        return ""
+
+
+KUBELET_NODE_ENV_VAR = 'DD_KUBERNETES_KUBELET_NODENAME'
+HOSTNAME_ENV_VAR = 'HOSTNAME'
+CAPACITY_ANNOTATION_KEY = 'CapacityProvisioned'
+POD_LIST_PATH = '/pods'
+
+log = logging.getLogger('collector')
 
 
 class EksFargateCheck(AgentCheck):
     """
-    Generate a heartbeat for Amazon EKS on AWS Fargate workloads
+    Generate heartbeat and capacity metrics for Amazon EKS on AWS Fargate workloads
     """
 
     def __init__(self, name, init_config, instances):
         super(EksFargateCheck, self).__init__(name, init_config, instances)
-        pod_name = os.getenv("HOSTNAME")
-        virtual_node = os.getenv("DD_KUBERNETES_KUBELET_NODENAME", "")
+        self.NAMESPACE = 'eks.fargate'
+
+        pod_name = os.getenv(HOSTNAME_ENV_VAR)
+        virtual_node = os.getenv(KUBELET_NODE_ENV_VAR, "")
+        self.fargate_mode = 'fargate' in virtual_node and pod_name is not None
 
-        self.fargate_mode = 'fargate' in virtual_node
+        kubelet_conn_info = get_connection_info()
+        endpoint = kubelet_conn_info.get('url')
+        if endpoint is None:
+            raise CheckException("Unable to detect the kubelet URL automatically: " + kubelet_conn_info.get('err', ''))
 
-        if pod_name is not None and self.fargate_mode:
+        self.pod_list_url = endpoint.strip("/") + POD_LIST_PATH
+        self.kubelet_credentials = KubeletCredentials(kubelet_conn_info)
+        self.pod_list = self.get_pod_list()
+
+        if self.fargate_mode:
             self.tags = []
-            self.tags.append("pod_name:" + pod_name)
-            self.tags.append("virtual_node:" + virtual_node)
+            self.tags.append('pod_name:' + pod_name)
+            self.tags.append('virtual_node:' + virtual_node)
             self.tags.extend(instances[0].get('tags', []))
 
     def check(self, instance):
-
-        # Only submit the heartbeat metric for fargate virtual nodes.
         if self.fargate_mode:
-            self.gauge("eks.fargate.pods.running", 1, self.tags)
+            for pod in self.pod_list.get('items', []):
+                pod_id = pod.get('metadata', {}).get('uid')
+                tags = tagger.tag('kubernetes_pod_uid://%s' % pod_id, tagger.ORCHESTRATOR) or []
+                # Submit the heartbeat metric for fargate virtual nodes.
+                self.gauge(self.NAMESPACE + '.pods.running', 1, set(self.tags + tags))
+                pod_annotations = pod.get('metadata', {}).get('annotations')
+                if CAPACITY_ANNOTATION_KEY not in pod_annotations:
+                    continue
+                cpu_val, mem_val = extract_resource_values(pod_annotations.get(CAPACITY_ANNOTATION_KEY))
+                if cpu_val == 0 and mem_val == 0:
+                    continue
+                self.gauge(self.NAMESPACE + '.capacity.cpu', cpu_val, set(self.tags + tags))
+                self.gauge(self.NAMESPACE + '.capacity.memory', mem_val, set(self.tags + tags))
```

Comment:
```suggestion
                self.gauge(self.NAMESPACE + '.cpu.capacity', cpu_val, set(self.tags + tags))
                self.gauge(self.NAMESPACE + '.memory.capacity', mem_val, set(self.tags + tags))
```
no strong opinion here, just to be aligned with kubelet metrics (there is a `kubernetes.memory.capacity` metric already)

Reply:
good call, i checked with yair and we will go with this instead